packages/plugins/src/plugins/sushiswap/farmingFetcher.ts

```diff
@@ -0,0 +1,153 @@
+import { Address, getAddress } from 'viem';
+import { Cache } from '../../Cache';
+import { Fetcher, FetcherExecutor } from '../../Fetcher';
+import { getEvmClient } from '../../utils/clients';
+import { PLATFORM_ID, FARMING_POOLS_KEY, SUSHI_TOKEN } from './constants';
+import { farmingAbi } from './abis';
+import { FarmingPool } from './types';
+import { ElementRegistry } from '../../utils/elementbuilder/ElementRegistry';
+import { zeroBigInt } from '../../utils/misc/constants';
+
+/**
+ * This fetcher fetches the MasterChef V1 and V2 pools to get the balances of the LP tokens and the rewards.
+ * @param networkId - The network ID
+ * @returns The fetcher
+ */
+export function getFarmingFetcher(networkId: EvmNetworkIdType): Fetcher {
+  const executor: FetcherExecutor = async (
+    owner: string,
+    cache: Cache
+  ): Promise<PortfolioElement[]> => {
+    const pools = await cache.getItem<FarmingPool[]>(FARMING_POOLS_KEY, {
+      prefix: PLATFORM_ID,
+      networkId,
+    });
+    if (!pools?.length) return [];
+
+    const client = getEvmClient(networkId);
+    const elementRegistry = new ElementRegistry(networkId, PLATFORM_ID);
+    const ownerAddress = getAddress(owner);
+
+    // Group pools by MasterChef address
+    const poolsByMasterChef = pools.reduce((acc, pool) => {
+      if (!acc[pool.masterChefAddress]) {
+        acc[pool.masterChefAddress] = [];
+      }
+      acc[pool.masterChefAddress].push(pool);
+      return acc;
+    }, {} as { [key: Address]: FarmingPool[] });
+
+    // Process each MasterChef's pools
+    for (const [masterChefAddress, masterChefPools] of Object.entries(
+      poolsByMasterChef
+    ) as [Address, FarmingPool[]][]) {
+      const poolsWithRewarders = masterChefPools.filter(
+        (pool) => pool.rewarder
+      );
+
+      const userInfoCalls = masterChefPools.map((pool) => ({
+        address: masterChefAddress,
+        abi: farmingAbi,
+        functionName: 'userInfo',
+        args: [BigInt(pool.pid), ownerAddress],
+      }));
+
+      const pendingSushiCalls = masterChefPools.map((pool) => ({
+        address: masterChefAddress,
+        abi: farmingAbi,
+        functionName: 'pendingSushi',
+        args: [BigInt(pool.pid), ownerAddress],
+      }));
+
+      const pendingTokensCalls = poolsWithRewarders.map((pool) => ({
+        address: pool.rewarder as Address,
+        abi: farmingAbi,
+        functionName: 'pendingTokens',
+        args: [BigInt(pool.pid), ownerAddress, zeroBigInt],
+      }));
+
+      const results = await client.multicall({
+        contracts: [
+          ...userInfoCalls,
+          ...pendingSushiCalls,
+          ...pendingTokensCalls,
+        ],
+      });
+
+      for (let i = 0; i < masterChefPools.length; i++) {
+        const pool = masterChefPools[i];
+        const userInfoRes = results[i];
+        const pendingSushiRes = results[i + masterChefPools.length];
+        const pendingTokensRes = pool.rewarder
```

Comment:
Doing this here will really clean up your types below. You should be able to get rid of all other type castinsg and your `unknown`

```suggestion
        const userInfoRes = results[i] as MulticallResult<YourType>;
        const pendingSushiRes = results[i + masterChefPools.length] as MulticallResult<YourType>;
        const pendingTokensRes = pool.rewarder as MulticallResult<YourType>
```